packages/postgres-database/src/simcore_postgres_database/models/services_limitations.py

```diff
@@ -0,0 +1,122 @@
+""" Groups table
+
+    - List of groups in the framework
+    - Groups have a ID, name and a list of users that belong to the group
+"""
+
+
+from typing import Final
+
+import sqlalchemy as sa
+from sqlalchemy import event
+
+from ._common import column_created_datetime, column_modified_datetime
+from .base import metadata
+from .clusters import clusters
+from .groups import groups
+
+_TABLE_NAME = "services_limitations"
+
+services_limitations = sa.Table(
+    _TABLE_NAME,
+    metadata,
+    sa.Column(
+        "gid",
+        sa.BigInteger,
+        sa.ForeignKey(
+            groups.c.gid,
+            onupdate="CASCADE",
+            ondelete="CASCADE",
+            name=f"fk_{_TABLE_NAME}_to_groups_gid",
+        ),
+        nullable=False,
+        doc="Group unique ID",
+    ),
+    sa.Column(
+        "cluster_id",
+        sa.BigInteger(),
+        sa.ForeignKey(
+            clusters.c.id,
+            name=f"fk_{_TABLE_NAME}_to_clusters_id",
+            onupdate="CASCADE",
+            ondelete="CASCADE",
+        ),
+        nullable=True,
+        doc="The cluster id with which these limitations are associated, if NULL or 0 uses the default",
+    ),
+    sa.Column(
+        "ram",
+        sa.BigInteger,
+        nullable=True,
+        doc="defines this group maximum allowable RAM used per service "
+        "(None means use defaults, <0 means no limits)",
+    ),
+    sa.Column(
+        "cpu",
+        sa.Numeric,
+        nullable=True,
+        doc="defines this group maximum allowable CPUs used per service "
+        "(None means use defaults, <0 means no limits)",
+    ),
+    sa.Column(
+        "vram",
+        sa.BigInteger,
+        nullable=True,
+        doc="defines this group maximum allowable VRAM used per service "
+        "(None means use defaults, <0 means no limits)",
+    ),
+    sa.Column(
+        "gpu",
+        sa.Integer,
+        nullable=True,
+        doc="defines this group maximum allowable CPUs used per service "
+        "(None means use defaults, <0 means no limits)",
+    ),
+    # TIME STAMPS ----
+    column_created_datetime(timezone=True),
+    column_modified_datetime(timezone=True),
+    sa.UniqueConstraint(
+        "gid",
+        "cluster_id",
+        name="gid_cluster_id_uniqueness",
+    ),
+    # prevents having multiple entries with NULL cluster (postgres < 15 treats NULL as always different)
+    sa.Index(
+        "idx_unique_gid_cluster_id_null",
+        "gid",
+        unique=True,
+        postgresql_where=sa.text("cluster_id IS NULL"),
+    ),
+)
+
+
+# TRIGGERS ------------------------
```

Comment:
So this is to produce the proper timestamp for update, right?
It is a bit convoluted. Do you think we can move it to  `_common` module so we can reuse it?

Reply:
well I guess we could re'use the procedure since we have the same attributes almost everywhere. otherwise the trigger is scoped to the table.
but we can check together tomorrow if you want. I don't see right away how to go

Reply:
done